api/envoy/config/filter/network/dubbo_proxy/v2alpha1/dubbo_proxy.proto

```diff
@@ -1,27 +1,54 @@
 syntax = "proto3";
 
-package envoy.extensions.filters.network.dubbo_proxy.v2alpha1;
-option java_package = "io.envoyproxy.envoy.extensions.filters.network.dubbo_proxy.v2alpha1";
+package envoy.config.filter.network.dubbo_proxy.v2alpha1;
+option java_package = "io.envoyproxy.envoy.config.filter.network.dubbo_proxy.v2alpha1";
 option go_package = "v2";
 
+import "envoy/config/filter/network/dubbo_proxy/v2alpha1/route.proto";
+
+import "google/protobuf/any.proto";
+
 import "validate/validate.proto";
+import "gogoproto/gogo.proto";
 
 // [#protodoc-title: Dubbo Proxy]
 // Dubbo Proxy filter configuration.
-
 message DubboProxy {
   // The human readable prefix to use when emitting statistics.
   string stat_prefix = 1 [(validate.rules).string.min_bytes = 1];
 
   // Configure the protocol used.
-  enum ProtocolType {
-    Dubbo = 0; // the default protocol.
-  }
   ProtocolType protocol_type = 2 [(validate.rules).enum.defined_only = true];
 
   // Configure the serialization protocol used.
-  enum SerializationType {
-    Hessian2 = 0; // the default serialization protocol.
-  }
   SerializationType serialization_type = 3 [(validate.rules).enum.defined_only = true];
+
+  // The route table for the connection manager is static and is specified in this property.
+  repeated RouteConfiguration route_config = 4;
+
+  // A list of individual Dubbo filters that make up the filter chain for requests made to the
+  // Dubbo proxy. Order matters as the filters are processed sequentially. For backwards
+  // compatibility, if no dubbo_filters are specified, a default Dubbo router filter
+  // (`envoy.filters.dubbo.router`) is used.
+  repeated DubboFilter dubbo_filters = 5;
```

Comment:
@gengleilei I am new to dubbo, my question is  how to setup` individual Dubbo filters`? 

I can not find any other supported filters except `envoy.filters.dubbo.router` .

Reply:
You can refer to the Router (router_impl.h), you need to implement DubboFilters: : DecoderFilter interface, yaml configuration is as follows:    
    stat_prefix: ingress
    route_config:
      name: local_route
    dubbo_filters:
      - name: envoy.filters.dubbo.mock_filter
        config:
          "@type": type.googleapis.com/google.protobuf.Struct
          value:
            name: test_service

Reply:
IC， this is for advanced users, need to hack into the envoy code.

Reply:
The Filter based extension mechanism is similar to HTTP. Dubbo does not define a unique Filter mechanism. of course, if you have good suggestions, we are also very welcome to build together.

Reply:
I got this error, any idea?
 ```
[2019-08-06 02:35:19.872][27][error][dubbo] [external/envoy/source/extensions/filters/network/dubbo_proxy/conn_manager.cc:132] [C35] dubbo error: invalid dubbo mes
sage magic number 20186
```